rust/crd/src/lib.rs

```diff
@@ -353,7 +353,6 @@ impl Storage {
 #[serde(rename_all = "kebab-case")]
 #[strum(serialize_all = "kebab-case")]
 pub enum Container {
-    Prepare,
```

Comment:
This is actually an breaking CRD change I guess, as you can not configure logging on the prepare container any more.
Is that correct? ;) In that case we should mark it as BREAKING and provide a migration path (delete .spec.foo.logging.prepare)

Reply:
Good point but its not actually breaking (Hashmap -> no CRD change). I grepped through the code to check for any left over "prepare" elements. If you set values for the prepare container they will just be ignored.
Should i explicitly mention that the prepare container was removed in the changelog?

Reply:
I would have assumed that the valid keys are only the known container as of, rendering currently valid CRs invalid.
https://github.com/stackabletech/operator-rs/blob/b3531bc2922f71f691807053e174404db95f1fb7/src/product_logging/spec.rs#L79

Actually tested it:
```
        apiVersion: kafka.stackable.tech/v1alpha1
        kind: KafkaCluster
        metadata:
          name: simple-kafka
        spec:
          image:
            productVersion: 3.4.0
          clusterConfig:
            zookeeperConfigMapName: xyz
          brokers:
            config:
              logging:
                enableVectorAgent: true
                containers:
                  foo:
                    level: DEBUG
            roleGroups:
              default: {}
```
thread 'tests::test_client_tls' panicked at 'illegal test input: Error("spec.brokers: unknown variant `foo`, expected one of `prepare`, `vector`, `kcat-prober`, `get-service`, `kafka`", line: 12, column: 13)', rust/crd/src/lib.rs:519:63

Reply:
We *could* keep the enum variant and ignore it for now, but honestly I would just remove it, as probably no one is using it right now and automatic CRD conversion is pretty far away.

Reply:
Not a fan of keeping stuff that does not do anything :)